ml-proto/spec/check.ml

```diff
@@ -134,9 +134,9 @@ let rec check_expr c et e =
     check_expr_opt c (label c x) eo e.at
 
   | Br_if (x, eo, e) ->
+    check_expr_opt c et eo e.at;
```

Comment:
@sunfishcode, that's not correct, the two expressions are completely different. The latter won't transfer `arg` to `$l`, and thus won't type-check if `$l` actually requires a value. There is no trivial way to rewrite the former under the current PR. You'll essentially need the encoding of a "drop" operator, avoiding which was a core design goal of the type system. (The simplest such encoding I can see would be `(block (if (i32.const 1) (br_if $l (arg) (cond)) (other))`.)

@lukewagner, special-casing typing on label arity would paint us into a corner regarding multiple values. I think we need a more future-proof solution. Reverting #180 is the one I can see.

I'm starting to wish we had a let-like construct, then we could easily desugar `(br_if $l (arg) (cond))` as follows and get the desired typing for free:
```
(let $x (arg) (block (if (cond) (br $l (get_value $x))) (get_value $x)))
```
:)